api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
+@groups.route('/manager-types/<int:manager_type_id>', methods=['PATCH'])
 @jwt_required
-def create_meeting():
-    if 'active' not in request.json.keys():
-        request.json['active'] = True
+def update_manager_type(manager_type_id):
+    manager_type_schema = ManagerTypeSchema()
+
     try:
-        valid_meeting = meeting_schema.load(request.json)
+        valid_attributes = manager_type_schema.load(request.json, partial=True)
     except ValidationError as err:
         return jsonify(err.messages), 422
 
-    new_meeting = Meeting(**valid_meeting)
-    db.session.add(new_meeting)
-    db.session.commit()
-    return jsonify(meeting_schema.dump(new_meeting)), 201
-
+    return modify_entity(ManagerType, manager_type_schema, manager_type_id, valid_attributes)
 
-@groups.route('/meetings')
-def read_all_meetings():
-    result = db.session.query(Meeting).all()
 
-    if result == []:
-        return jsonify(msg="No meetings found"), 404
+@groups.route('/manager-types/<int:manager_type_id>', methods=['DELETE'])
+@jwt_required
+def delete_manager_type(manager_type_id):
+    manager_type = db.session.query(ManagerType).filter_by(id=manager_type_id).first()
 
-    for r in result:
-        r.address = r.address
-    return jsonify(meeting_schema.dump(result, many=True))
+    if not manager_type:
+        return jsonify(f"ManagerType with id #{manager_type_id} does not exist."), 404
 
+    db.session.delete(manager_type)
+    db.session.commit()
 
-@groups.route('/meetings/group/<group_id>')
-@jwt_required
-def read_all_meetings_by_group(group_id):
-    result = db.session.query(Meeting).filter_by(group_id=group_id).all()
+    # 204 codes don't respond with any content
+    return "Deleted successfully", 204
 
-    if len(result) == 0:
-        return jsonify(msg="No meetings found"), 200
 
-    for r in result:
-        r.address = r.address
+# ---- Manager
 
-    return jsonify(meeting_schema.dump(result, many=True))
+manager_schema = ManagerSchema()
 
 
-@groups.route('/meetings/address/<address_id>')
+@groups.route('/groups/<int:group_id>/managers', methods=['POST'])
 @jwt_required
-def read_all_meetings_by_location(address_id):
-    result = db.session.query(Meeting).filter_by(address_id=address_id).all()
+def create_manager(group_id):
+    # make group_id an invalid field in payload
+    manager_schema = ManagerSchema(exclude=['group_id'])
+    try:
+        valid_manager = manager_schema.load(request.json, partial=['active'])
+    except ValidationError as err:
+        return jsonify(err.messages), 422
 
-    if len(result) == 0:
-        return jsonify(msg="No meetings found"), 404
+    person_id = valid_manager['person_id']
 
-    return jsonify(meeting_schema.dump(result, many=True))
+    if db.session.query(Manager).filter_by(person_id=person_id, group_id=group_id).first():
+        # if the same manager already exists
+        return jsonify(f"Manager with group_id #{group_id} and person_id #{person_id} already exists"), 409
```

Comment:
This error message is not going to be very helpful for the end user. Rather than talking about database IDs that have no meaning to the user, could this be changed to say something like "Fred Ziffle is already a manager of the Young Families group."

Reply:
Yeah. That makes total sense. And like the `jsonify(...some text ...)` construct, I think this is also a common practice in the code base since a long time ago. Do we want to combine those problems into an issue and make change to all of them at once?

Reply:
Yes, let's create an issue (if you've not done so already). This PR is not the right place to resolve the question.